From 099652e7cff8ece51afc4f2690409d3db7afe8e1 Mon Sep 17 00:00:00 2001
From: ninez <johnstonljordan@gmail.com>
Date: Sat, 29 Jan 2022 09:59:17 -0500
Subject: [PATCH 210/228] designware-i2c IRQF_NO_THREAD

 - use IRQF_NO_THREAD in the designware-i2c driver, this doesn't completely fix
   my woes, but it's better than leaving it threaded.

Signed-off-by: ninez <johnstonljordan@gmail.com>
Signed-off-by: Jordan Johnston <johnstonljordan@gmail.com>
---
 drivers/i2c/busses/i2c-designware-master.c | 4 ++--
 drivers/i2c/busses/i2c-designware-slave.c  | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-designware-master.c b/drivers/i2c/busses/i2c-designware-master.c
index 44a94b225ed8..bff5f38bcef9 100644
--- a/drivers/i2c/busses/i2c-designware-master.c
+++ b/drivers/i2c/busses/i2c-designware-master.c
@@ -895,9 +895,9 @@ int i2c_dw_probe_master(struct dw_i2c_dev *dev)
 		return amd_i2c_adap_quirk(dev);
 
 	if (dev->flags & ACCESS_NO_IRQ_SUSPEND) {
-		irq_flags = IRQF_NO_SUSPEND;
+		irq_flags = IRQF_NO_SUSPEND | IRQF_NO_THREAD;
 	} else {
-		irq_flags = IRQF_SHARED | IRQF_COND_SUSPEND;
+		irq_flags = IRQF_SHARED | IRQF_COND_SUSPEND | IRQF_NO_THREAD;
 	}
 
 	ret = i2c_dw_acquire_lock(dev);
diff --git a/drivers/i2c/busses/i2c-designware-slave.c b/drivers/i2c/busses/i2c-designware-slave.c
index 0d15f4c1e9f7..b18b7b71b0ff 100644
--- a/drivers/i2c/busses/i2c-designware-slave.c
+++ b/drivers/i2c/busses/i2c-designware-slave.c
@@ -276,7 +276,7 @@ int i2c_dw_probe_slave(struct dw_i2c_dev *dev)
 	i2c_set_adapdata(adap, dev);
 
 	ret = devm_request_irq(dev->dev, dev->irq, i2c_dw_isr_slave,
-			       IRQF_SHARED, dev_name(dev->dev), dev);
+			       IRQF_SHARED | IRQF_NO_THREAD, dev_name(dev->dev), dev);
 	if (ret) {
 		dev_err(dev->dev, "failure requesting irq %i: %d\n",
 			dev->irq, ret);
-- 
2.36.1

